yarelismartin/Bookworm-Society-BE
# Voting session endpoints

﻿## Detailed Description


Added all the endpoints needed in order for voting sessions to work. They all have error handling to deal with situations where a user cant vote in that club or maybe has already voted. I added a background service which is essentially a long-running task that runs independently of the request-response lifecycle that we are use to using with our HTTP requests. 

## Related Issues
#30 
#31
 #19
#32 
#29 

## How to Test


1.  Go into your pgAdmin and create a voting session. Make sure to select books to be voted on by adding it to the respective table. 
2. In the voest table add some votes that correspond to the ids of the books being voteD on. 
3. After you've done all that this is what you need to test.
- The booksclubs book id get updated with the book with the most votes. 
- The have read gets a new row with the ids to the bookclub and it's previoud book obj id
- the voting session goes from isActive: true to false. It's winningBookId gets updated with the book that won
4. After this you can also check scenarioes where their is a tie. If this happens the books get selected at random. 
5. Lastly check what happens if there are no votes. Currently I have it set up to select the first book in the books to vote on list.



## Screenshots (if applicable)

## Types of Changes

- [ x] ✨ New feature

## 💬 Additional Notes
